### DISCUSSION TITLE
Iphone IKEv2 certificate auth - "received DELETE" at rekey

### DISCUSSION THREAD

**Discussion by t-dubbb** [strongswan/strongswan#1578] [Q&A]
Trying to troubleshoot why an iPhone is sending a disconnect immediately after tunnel rekeying, apparently this happens within 8 minutes? The same server certificate was installed on a Fortigate firewall with the same Iphone certificate and it ran beautifully, really don't think it's a cert problem. Below is ipsec.conf and a log during a disconnect.

> 16[NET] received packet: from IPHONE_IP[11209] to SERVER_IP[4500] (416 bytes)
> 16[ENC] parsed CREATE_CHILD_SA request 7 [ SA No KE ]
> 16[IKE] IPHONE_IP is initiating an IKE_SA
> 12[MGR] ignoring request with ID 7, already processing
> 16[IKE] IKE_SA RA[25] rekeyed between SERVER_IP[SERVER_CERT_DN]...IPHONE_IP[IPHONE_CERT_DN]
> 16[ENC] generating CREATE_CHILD_SA response 7 [ SA No KE ]
> 16[NET] sending packet: from SERVER_IP[4500] to IPHONE_IP[11209] (432 bytes)
> 07[NET] received packet: from IPHONE_IP[11209] to SERVER_IP[4500] (80 bytes)
> 07[ENC] parsed INFORMATIONAL request 8 [ D ]
> 07[IKE] received DELETE for IKE_SA RA[2]
> 07[IKE] deleting IKE_SA RA[2] between SERVER_IP[SERVER_CERT_DN]...IPHONE_IP[IPHONE_CERT_DN]
> 07[IKE] IKE_SA deleted
> 07[ENC] generating INFORMATIONAL response 8 [ ]
> 07[NET] sending packet: from SERVER_IP[4500] to IPHONE_IP[11209] (80 bytes)

> conn RA
>     keyexchange=ikev2
>     ike=aes256-sha256-modp2048!
>     esp=aes256-sha256!
>     reauth=no
>     rekey=no
>     forceencaps=yes
>     dpdaction=restart
>     dpddelay=60s
>     dpdtimeout=300s
>     left=%any
>     leftca=/config/auth/ca.crt
>     leftauth=pubkey
>     leftid=@SERVER_CERT_DN
>     leftsubnet=0.0.0.0/0
>     leftcert=/config/auth/Home.crt
>     leftsendcert=always
>     right=%any
>     rightauth=pubkey
>     rightsourceip=172.16.16.64/29
>     rightdns=172.16.16.2
>     rightid=%any
>     rightca=%same
>     rightsendcert=never
>     auto=add

**Comment by tobiasbrunner** [comment]
No idea, you may want to try checking the client log (I think it's accessible via Xcode).
